inc/Custom_Menu_Items.php

```diff
@@ -243,7 +245,7 @@ public function nav_menu_item_fields( $nav_menu_item_fields, $context ) {
 				$new_nav_menu_item_fields['description'] = $nav_menu_item_fields['description'];
 			}
 
-			if ( 'newsletter_box' === $context['item']->rcmit_type ) {
+			if ( 'shortcode_box' === $context['item']->rcmit_type || 'newsletter_box' === $context['item']->rcmit_type ) {
```

Comment:
How about adding this below `if ( ! empty( $context['item']->rcmit_type ) ) {`:

```php
if ( ! empty( $context['item']->rcmit_type ) ) {
    // Migrate old type to new on-the-fly.
	if ( 'newsletter_box' === $context['item']->rcmit_type ) {
		$context['item']->rcmit_type = 'shortcode_box';
    }
```

With that you don't have to check both types here.

Reply:
In my next PR I am replacing the if statements with a switch case.

Reply:
My point is more the migration to the new type which will happen when the menu gets saved.